processor/src/main/java/com/ihren/processor/validation/discount/ValidDiscount.java

```diff
@@ -0,0 +1,19 @@
+
+package com.ihren.processor.validation.discount;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD, ElementType.PARAMETER })
+@Constraint(validatedBy = DiscountStringValidator.class)
+public @interface ValidDiscount {
```

Comment:
It is not necessary to have Valid prefix for annotation which is already for validation. It can be called just Discount